lib/rails-mermaid_erd_markdown.rb

```diff
@@ -14,59 +14,51 @@ module MermaidErdMarkdown
   class << self
     attr_writer :logger
 
-    def logger
-      @logger ||= Logger.new($stdout).tap do |log|
-        log.progname = name
-      end
-    end
-
-    def configuration
-      @configuration ||= MermaidErdMarkdown::Configuration.new
-    end
-
     def perform
-      new_erd = generate_mermaid_erd
-      existing_erd_path = Pathname.new(configuration.output_path)
-
-      unless existing_erd_path.exist?
-        logger.info("ERD does not currently exist at result path. Creating...")
-        begin
-          create_erd_file(new_erd)
-          logger.info("ERD successfully created at #{configuration.output_path}")
-        rescue StandardError
-          logger.info("Could not create ERD. Output path is invalid.")
-        end
+      return unless find_or_create_output
+      return unless erd_changed?
 
-        return
-      end
+      update_erd
 
-      existing_erd = File.read(existing_erd_path)
-      update_erd_file(existing_erd, new_erd)
+      update_split_erds if configuration.split_output
     end
 
-    def update_erd_file(current_erd, new_erd)
-      # check if two diagrams are the same by comparing their MD5 hashes
-      if Digest::MD5.hexdigest(current_erd) == Digest::MD5.hexdigest(new_erd)
-        logger.info("ERD already exists and is up to date. Skipping...")
-        return
+    def index_markdown(files)
+      lines = ["# Entity Relationship Diagrams"]
+      lines << ""
+      lines << "Each model has its own ERD diagram. The diagram shows the "\
+        "selected model, plus #{configuration.relationship_depth} "\
+        "associated model(s) deep. Click on the model name to view "\
+        "the diagram."
+      lines << ""
+      lines << "## Models"
+      lines << ""
+      files.each do |model, path|
+        lines << "- [#{model}](#{path})"
```

Comment:
The index ERD file is also in the top level of the path, which means that this path won't work. 
Prepending a `../` should do it.

Reply:
This should be okay since I am appending an `_index` to the filename.

Reply:
Oh... it is outputting `docs/thing` and that doesn't link from `docs/index`, got it!